src/poller.js

```diff
@@ -87,7 +87,14 @@ module.exports = EventEmitter => {
 			// later is discarded within pollers
 			const _fetch = this.options.retry ? this.eagerFetch : fetch;
 
-			return _fetch (this.url, this.options)
+			const options = {...this.options}
+			if (options.timeout) {
+				// add signal option to support native fetch, but keep timeout option
+				// too to support node-fetch@<2.3.0
+				options.signal = AbortSignal.timeout(options.timeout)
+			}
```

Comment:
**suggestion**: think we should also deprecate the `timeout` option so we can eventually go back to transparently passing the options through to `fetch`, wdyt?

Reply:
Oh that's quite nice - log a warning?

Reply:
not sure that's possible actually: we have to recreate the `AbortSignal` each time in the `.fetch` method as the timeout starts ticking down from when it's created, not when the `fetch` starts. so we can't pass the same `signal` argument from the constructor to `fetch` each time!

Reply:
ohhh, fair enough